src/components/slide/preview-deck.tsx

```diff
@@ -35,8 +36,8 @@ export const PreviewDeck: React.FC = () => {
 
   useEffect(() => {
     try {
-      if (!slideTemplateJson) {
-        return;
+      if (!slideTemplateJson || slideTemplateJson.children.length === 0) {
+        return navigate(PATHS.VISUAL_EDITOR);
```

Comment:
Since this is in a `useEffect`:
```suggestion
        navigate(PATHS.VISUAL_EDITOR);
        return;
```

Reply:
Yes, good catch. Thank you!